systers_portal/templates/base.html

```diff
@@ -197,11 +203,24 @@
     </p>
   </div>
 </div>
-</div>
-<script src="{% static 'js/libs/jquery-1.11.1.min.js' %}"></script>
+
+<!-- <script src="{% static 'js/libs/jquery-1.11.1.min.js' %}"></script> -->
+
 <script src="{% static 'js/libs/bootstrap.min.js' %}"></script>
 <script src="{% static 'js/choose_profile_pic.js' %}"></script>
 <script src="{% static 'js/search_menu.js' %}"></script>
+<script>
+  $(document).ready(()=>{
+      const optionlength = $('select > option').length;
+      if(optionlength>1500){
+        $('select').select2({
+          minimumInputLength:1
+        });
+      }else{
+        $('select').select2();
+      }
+  });
+</script>
```

Comment:
I understand the temptation to use jquery, but please look into something much more lightweight here like https://github.com/jshjohnson/Choices

Also, http://youmightnotneedjquery.com/

Reply:
Regarding the size Select2 has 65k and choice has 19kb. I don't use it, but in the choice's repo issue I noticed that it is not good for large dataset.
[See](https://github.com/jshjohnson/Choices/issues/833)

Reply:
You could take it for a spin and see how it works.

Reply:
I think select2 serves our requirements here better than choices, also size difference not much more. Also, choices are no longer maintained.
Here the dataset contains>1500 options.
What do you think have you use choices previously? Have you tried it locally?

Reply:
I have not tested either of these, these are merely suggestions.

I would also like you to check generic autocomplete libraries as they are much more lightweight but do not provide select integration.

If you think select2 is fine, this looks good to me otherwise.